### PR TITLE
feat: run end bool compressor

### PR DESCRIPTION
**Pull request by lwwmanning**

### CODE REVIEW

File: vortex-sampling-compressor/src/compressors/runend_bool.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        RunEndBool::ID.as_ref()
+    }
+
+    fn cost(&self) -> u8 {
+        constants::RUN_END_BOOL_COST
+    }
+
+    fn can_compress(&self, array: &ArrayData) -> Option<&dyn EncodingCompressor> {
+        // Only support bool arrays
+        if array.encoding().id() != Bool::ID {
```
Comment (robert3005) [inline vortex-sampling-compressor/src/compressors/runend_bool.rs]:
```suggestion
        if !array.is_encoding(Bool::ID) {
```

File: encodings/runend-bool/src/array.rs [hunk 2]
```diff
@@ -146,6 +147,36 @@ impl RunEndBoolArray {
...
+    })
+}
+
+pub(crate) fn value_at_index(idx: usize, start: bool) -> bool {
+    if idx % 2 == 0 {
+        start
+    } else {
+        !start
+    }
+}
```
Comment (robert3005) [inline encodings/runend-bool/src/array.rs:178]: don't quite get this move. I think it's nicer if it's in one place?

**Review by robert3005 — APPROVED** [review]

File: encodings/runend-bool/src/array.rs (hunk 2 above)
Comment (lwwmanning) [reply encodings/runend-bool/src/array.rs:178]: called in a bunch of places all over the crate (including in compute), felt silly for it to live in the `compress` module

Comment (robert3005) [reply encodings/runend-bool/src/array.rs:178]: ok, I just looked at the diff

**Pull request merged**